// poe-api-manager/dist/lib/modules/poe.watch/WatchViewModule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WatchViewModule = void 0;
const Modules_1 = require("./allModules/Modules");
/**
 * Represents a module for fetching data related to Watchstones and Watchstone Regions.
 */
class WatchViewModule {
    /**
     * Module for fetching data related to Currency.
     */
    currency;
    /**
     * Module for fetching data related to Accessories.
     */
    accessory;
    /**
     * Module for fetching data related to Armour.
     */
    armour;
    /**
     * Module for fetching data related to Base.
     */
    base;
    /**
     * Module for fetching data related to Beast.
     */
    beast;
    /**
     * Module for fetching data related to DeliriumOrb.
     */
    deliriumOrb;
    /**
     * Module for fetching data related to Divs.
     */
    divs;
    /**
     * Module for fetching data related to Essence.
     */
    essence;
    /**
     * Module for fetching data related to Fossil.
     */
    fossil;
    /**
     * Module for fetching data related to Fragment.
     */
    fragment;
    /**
     * Module for fetching data related to Map.
     */
    map;
    /**
     * Module for fetching data related to Flask.
     */
    flask;
    /**
     * Module for fetching data related to Jewel.
     */
    jewel;
    /**
     * Module for fetching data related to Gem.
     */
    gem;
    /**
     * Module for fetching data related to Inscribed.
     */
    //public inscribed: InscribedModule;
    /**
     * Module for fetching data related to Invitation.
     */
    invitation;
    /**
     * Module for fetching data related to Oil.
     */
    oil;
    /**
     * Module for fetching data related to Scarab.
     */
    scarab;
    /**
     * Module for fetching data related to Sextant.
     */
    sextant;
    /**
     * Module for fetching data related to UniqueMaps.
     */
    uniqueMaps;
    /**
     * Module for fetching data related to Weapon.
     */
    weapon;
    /**
     * Creates an instance of WatchWievModule.
     * @param league - The league name to fetch data from.
     */
    constructor(league) {
        this.currency = new Modules_1.CurrencyModule(league);
        this.accessory = new Modules_1.AccessoryModule(league);
        this.armour = new Modules_1.ArmourModule(league);
        this.base = new Modules_1.BaseModule(league);
        this.beast = new Modules_1.BeastModule(league);
        this.essence = new Modules_1.EssenceModule(league);
        this.fossil = new Modules_1.FossilModule(league);
        this.fragment = new Modules_1.FragmentModule(league);
        this.map = new Modules_1.MapModule(league);
        this.flask = new Modules_1.FlaskModule(league);
        this.jewel = new Modules_1.JewelModule(league);
        this.gem = new Modules_1.GemModule(league);
        this.invitation = new Modules_1.InvitationModule(league);
        this.oil = new Modules_1.OilModule(league);
        this.scarab = new Modules_1.ScarabModule(league);
        this.sextant = new Modules_1.SextantModule(league);
        this.weapon = new Modules_1.WeaponModule(league);
        this.uniqueMaps = new Modules_1.UniqueMapsModule(league);
        this.deliriumOrb = new Modules_1.DeliriumOrbModule(league);
        this.divs = new Modules_1.DivsModule(league);
        //this.inscribed = new InscribedModule(league);- not now
    }
}
exports.WatchViewModule = WatchViewModule;
exports.default = WatchViewModule;
